content/en/platform/corda/4.12/enterprise/node/operating/tvu/tvu-cli.md

```diff
@@ -0,0 +1,114 @@
+---
+date: '2023-12-15'
+section_menu: corda-enterprise-4-12
+menu:
+  corda-enterprise-4-12:
+    identifier: corda-enterprise-4-12-tvu-cli
+    parent: corda-enterprise-4-12-tvu
+tags:
+- tvu cli
+- tvu
+- transaction validator utility
+title: Transaction Validator Utility CLI parameters
+weight: 200
+---
+
+# Transaction Validator Utility CLI parameters
+
+The following section lists all the Transaction Validator Utility (TVU) CLI parameters. You can use them to perform various actions on transactions present in your database.
+
```

Comment:
Also mention these parameters are descried in the tool via the --help option.  @suhas-srivastava Is this correct.

For all the options can you specify what happens when no option is specified, some are done but not all e.g. cordapp-dir @suhas-srivastava

Reply:
Yes, `-h` or `--help` option gives info about all the CLI options.

Regarding `--cordapp-dir` CLI option:  `--cordapp-dir` CLI option is used to load cordapps into the utility to enable full transaction deserialisation. Transaction deserialisation is enabled whenever cordapps are provided to the utility and is disabled when no cordapps are provided. There are three cases in absence of this option:
1. If `--cordapp-dir` is not given and database connection is provided using the `-d` (or `--datasource`) CLI option then no cordapps are loaded and hence transaction deserialisation is _disabled_.
2. If `--cordapp-dir` is not given and node's configuration is provided using `-b` (or `--base-directory`) CLI option then cordapps present only on the node-base-directory's cordapp-directory are loaded, these loaded cordapps _enable_ transaction deserialisation.
3. If `--cordapp-dir` is not given and if `-b` and `-d` options are absent as well, then as per `-b` option's default configuration the current directory is treated as the node's base-directory. In this case all the cordapps contained within the current directory's cordapps directory are loaded and hence transaction deserialisation is _enabled_.

If both, node's-base-directory (see `-b` option configuration) and cordapps directory (--cordapp-dir) are provided then cordapps are loaded from both these sources and transaction deserialisation is _enabled_.

Reply:
updated